### PR TITLE
DashR DAQ IV Tracer

### PR DESCRIPTION
**Pull request by CanerIrfanoglu**
Issue for app: #[issue number here]

# App pull request
- [x] This is a new app
- [ ] I am improving an existing app (redesigns/code "makeovers")

## About

- Playground deployment URL (new version):
- Current gallery app URL: https://dash-gallery.plotly.host/dash-daq-iv-tracer/
- Python app repository link: https://github.com/plotly/dash-sample-apps/tree/master/apps/dash-daq-iv-tracer

## Workflow

- [x] I have created a branch in the appropriate monorepo, and the
  elements necessary for successful deployment are in place.
- [ ] If the app is a redesigned and/or restyled version of an
  existing gallery app, I've summarized the changes requested in the
  appropriate Streambed issue and confirm that they have been applied.
- [x] If the app is on the Dash Gallery portal, I have added a link to
  the GitHub repository for the source code in the portal description.
- [x] If the app is a reimplementation of a Python gallery app for the
  DashR gallery, the app in this PR mimics, as closely as possible,
  the style and functionality of the existing app.=
- [x] I have removed *all Google Analytics code* from the app's
  `assets/` folder.

## The pre-review review

I have addressed all of the following questions:

- [x] Does everything in my code serve some purpose? (I have removed
  any dead and/or irrelevant code.)
- [x] Does everything in my code have a clear purpose? (My code is
  readable and, where it isn't, it has been commented appropriately.)]
- [x] Am I reinventing the wheel? (I have used appropriate packages to
  lessen the volume of code that needs to be maintained.)

### CODE REVIEW

File: apps/dashr-daq-iv-tracer/app.R [hunk 1]
```diff
@@ -0,0 +1,1508 @@
...
+        plot_bgcolor = cardColor[[theme]],
+        paper_bgcolor = cardColor[[theme]]
+      )
+    )
+
+    if (!(modeChoice)) { # Single measure case
+      if (measTriggered) {
+
+        # Single data point does not draw graph,
+        # duplicating data to show single data point
```
Comment (rpkyle) [inline apps/dashr-daq-iv-tracer/app.R]: Interesting; why is this step necessary?

Comment (CanerIrfanoglu) [reply apps/dashr-daq-iv-tracer/app.R]: When the voltage/current value set from the `daqKnob` and `Single Measure` is clicked, we expect having a single data point on the graph like in the python version. However, when there's only 1 data point in `xdata` and `ydata`, the callback was returning an empty graph. Just duplicating values in this scenario solved the issue. However, not sure if it is the best solution.

Screen Shot 2019-08-08 at 2 55 07 PM

**Comment by CanerIrfanoglu** [comment]
@rpkyle thank you for reviewing. Just pushed commits https://github.com/plotly/dash-sample-apps/pull/275/commits/1e3e188e2c4ab12bb2dbc0fbdee0c4392ca63124 https://github.com/plotly/dash-sample-apps/pull/275/commits/d686966f94392c10f320b774632859eb5f762e74 applying the suggested changes.

File: apps/dashr-daq-iv-tracer/app.R (hunk 1 above)
Comment (rpkyle) [reply apps/dashr-daq-iv-tracer/app.R]: Ah, OK; thanks for the additional detail. I think we should revisit this sometime soon, but for now it's fine. We can discuss alternatives and then just open a new, tiny PR to update later.

**Comment by rpkyle** [comment]
Nice work!! 💃

**Comment by CanerIrfanoglu** [comment]
@KPhans thank you for reviewing :) Solid suggestions, definitely will look into  `callback_context`. Guess `DashR-oil-and-gas` would be a good example.

**Pull request merged**